### PR TITLE
Multi-region `aws_vpc` (vpc + subnet + az) and `aws_security_group`

### PR DESCRIPTION
**Pull request by aguillenv**
This will start unblocking the multi-region conversion of some modules

### CODE REVIEW

File: src/modules/0.0.22/aws_vpc/mappers/subnet.ts [hunk 1]
```diff
@@ -103,6 +103,13 @@ export class SubnetMapper extends MapperBase<Subnet> {
       // next loop through should delete the old one
       const out = [];
       for (const e of es) {
+        const cloudRecord =
+          ctx?.memo?.cloud?.Subnet?.[this.entityId(e)] ??
+          (await this.module.subnet.cloud.read(ctx, this.entityId(e)));
+        if (cloudRecord?.vpc?.vpcId === e.vpc?.vpcId) {
```
[the remark this answers is not in the document]
Comment (aguillenv) [reply src/modules/0.0.22/aws_vpc/mappers/subnet.ts:109]: oh, good catch! how does it work!? let me try again

**Review by yrobla — APPROVED** [review]

File: src/modules/0.0.22/aws_vpc/mappers/subnet.ts [hunk 2]
```diff
@@ -9,27 +9,28 @@ export class SubnetMapper extends MapperBase<Subnet> {
...
-  async subnetMapper(sn: AwsSubnet, ctx: Context) {
+  async subnetMapper(sn: AwsSubnet, ctx: Context, region: string) {
     const out = new Subnet();
     if (!sn?.SubnetId || !sn?.VpcId) return undefined;
     out.state = sn.State as SubnetState;
     if (!sn.AvailabilityZone) return undefined;
     out.availabilityZone =
-      (await this.module.availabilityZone.db.read(ctx, sn.AvailabilityZone)) ??
-      (await this.module.availabilityZone.cloud.read(ctx, sn.AvailabilityZone));
+      (await this.module.availabilityZone.db.read(ctx, `${sn.AvailabilityZone}|${region}`)) ??
```
Comment (depombo) [inline src/modules/0.0.22/aws_vpc/mappers/subnet.ts:22]: nit mentioned this in another PR, but having a static getter/setter function in the cloud id service that gets called here would be nice

File: test/modules/aws-vpc-integration.ts [hunk 3]
```diff
@@ -151,7 +152,7 @@ describe('VPC Integration Testing', () => {
       INSERT INTO nat_gateway (connectivity_type, subnet_id, tags)
       SELECT 'private', id, '{"Name":"${ng}"}'
       FROM subnet
-      WHERE cidr_block = '192.${randIPBlock}.0.0/16';
+      WHERE cidr_block = '191.${randIPBlock}.0.0/16';
```
Comment (depombo) [inline test/modules/aws-vpc-integration.ts:155]: why this change?

File: src/modules/0.0.22/aws_ecs_fargate/mappers/service.ts [hunk 4]
```diff
@@ -240,8 +245,10 @@ export class ServiceMapper extends MapperBase<Service> {
         // misconfigured resources
         let subnet: Subnet;
         try {
+          // todo: search by region when multiregion
```
[the remark this answers is not in the document]
Comment (dfellis) [reply src/modules/0.0.22/aws_ecs_fargate/mappers/service.ts:248]: I consider this part of the multiregion issue we are already tackling

File: test/modules/aws-vpc-integration.ts (hunk 3 above)
Comment (dfellis) [reply test/modules/aws-vpc-integration.ts:155]: For the new tests that update the cidr value further on, I believe. Changing from 191 to 192

File: src/modules/0.0.22/aws_vpc/mappers/subnet.ts (hunk 2 above)
Comment (dfellis) [reply src/modules/0.0.22/aws_vpc/mappers/subnet.ts:22]: Agreed

File: src/modules/0.0.22/aws_ecs_simplified/sql/after_install.sql [hunk 5]
```diff
@@ -98,7 +98,7 @@ BEGIN
   -- create ECS service and associate it to security group
   INSERT INTO service ("name", desired_count, subnets, assign_public_ip, cluster_name, task_definition_id, target_group_name, force_new_deployment)
   VALUES (
-    NEW.app_name || '-service', NEW.desired_count, (SELECT ARRAY(SELECT subnet_id FROM subnet WHERE vpc_id = (SELECT id FROM vpc WHERE is_default = true LIMIT 1) LIMIT 3)), (CASE WHEN NEW.public_ip THEN 'ENABLED' ELSE 'DISABLED' END)::service_assign_public_ip_enum,
+    NEW.app_name || '-service', NEW.desired_count, (SELECT ARRAY(SELECT subnet_id FROM subnet WHERE vpc_id = (SELECT id FROM vpc WHERE is_default = true and vpc.region = (SELECT region FROM aws_regions WHERE is_default = TRUE) LIMIT 1) LIMIT 3)), (CASE WHEN NEW.public_ip THEN 'ENABLED' ELSE 'DISABLED' END)::service_assign_public_ip_enum,
```
[the remark this answers is not in the document]
Comment (dfellis) [reply src/modules/0.0.22/aws_ecs_simplified/sql/after_install.sql:101]: 👍

File: src/modules/0.0.22/aws_vpc/mappers/subnet.ts (hunk 2 above)
Comment (aguillenv) [reply src/modules/0.0.22/aws_vpc/mappers/subnet.ts:22]: Well it would need to be like a helper like createCompositeKey or something customizable. In this case we couldn't used the entityId because the values come from different objects.

File: test/modules/aws-vpc-integration.ts (hunk 3 above)
Comment (aguillenv) [reply test/modules/aws-vpc-integration.ts:155]: It is also because the previous tests were kind of wrong. They were leaving behind the vpc with the 192 cidr and that's why the rest of the tests where working, but technically they should have failed
